js/binance.js

```diff
@@ -2244,6 +2244,45 @@ module.exports = class binance extends Exchange {
         return this.parseTickers (response, symbols);
     }
 
+    async fetchL1OrderBooks (symbols = undefined, params = {}) {
+        /**
+         * @method
+         * @name binance#fetchL1OrderBooks
+         * @description fetches the bid and ask price and volume for multiple markets
+         * @param {[str]|undefined} symbols unified symbols of the markets to fetch the bids and asks for, all markets are returned if not assigned
+         * @param {dict} params extra parameters specific to the binance api endpoint
+         * @returns {dict} an array of [ticker structures]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const defaultType = this.safeString2 (this.options, 'fetchBidsAsks', 'defaultType', 'spot');
+        const type = this.safeString (params, 'type', defaultType);
+        const query = this.omit (params, 'type');
+        let method = undefined;
+        if (type === 'future') {
+            method = 'fapiPublicGetTickerBookTicker';
+        } else if (type === 'delivery') {
+            method = 'dapiPublicGetTickerBookTicker';
+        } else {
+            method = 'publicGetTickerBookTicker';
+        }
+        const response = await this[method] (query);
+        return this.parseL1OrderBooks (response, symbols);
+    }
+
+    parseL1OrderBook (l1OrderBook, market = undefined) {
```

Comment:
This method is not defined in the base class, which probably should if we want to make it a generic parser to be overridden in derived files. If so, to have a more generic method's signature, I would suggest getting the`symbol` as a parameter for the cases where certain exchanges don't return it.

Reply:
you are right, but i think, at first, before defining it into base, we should collect more exchange-cases for this method, and once we have common patter we might define then, not? 
I think that (after this PR will be merged) i will go through other exchanges to add same fetch method, and once we have 4-5+ cases, we can then define in base.